tests/load/test_read_interfaces.py

```diff
@@ -0,0 +1,247 @@
+            ]
+
+        return [items]
+
+    # run source
+    s = source()
+    pipeline.run(s, loader_file_format=destination_config.file_format)
+
+    # access via key
+    table_relationship = pipeline.dataset()["items"]
+
+    # full frame
+    df = table_relationship.pandas()
+    assert len(df.index) == total_records
+
+    #
+    # check dataframes
+    #
+
+    # chunk
+    df = table_relationship.pandas(chunk_size=chunk_size)
+    if not skip_df_chunk_size_check:
+        assert len(df.index) == chunk_size
+    # lowercase results for the snowflake case
+    assert set(df.columns.values) == set(expected_columns)
+
+    # iterate all dataframes
+    frames = list(table_relationship.iter_pandas(chunk_size=chunk_size))
+    if not skip_df_chunk_size_check:
+        assert [len(df.index) for df in frames] == expected_chunk_counts
+
+    # check all items are present
+    ids = reduce(lambda a, b: a + b, [f[expected_columns[0]].to_list() for f in frames])
+    assert set(ids) == set(range(total_records))
+
+    # access via prop
+    table_relationship = pipeline.dataset().items
+
+    #
+    # check arrow tables
+    #
+
+    # full table
+    table = table_relationship.arrow()
+    assert table.num_rows == total_records
+
+    # chunk
+    table = table_relationship.arrow(chunk_size=chunk_size)
+    assert set(table.column_names) == set(expected_columns)
+    assert table.num_rows == chunk_size
+
+    # check frame amount and items counts
+    tables = list(table_relationship.iter_arrow(chunk_size=chunk_size))
+    assert [t.num_rows for t in tables] == expected_chunk_counts
+
+    # check all items are present
+    ids = reduce(lambda a, b: a + b, [t.column(expected_columns[0]).to_pylist() for t in tables])
+    assert set(ids) == set(range(total_records))
+
+    # check fetch accessors
+    table_relationship = pipeline.dataset().items
+
+    # check accessing one item
+    one = table_relationship.fetchone()
+    assert one[0] in range(total_records)
+
+    # check fetchall
+    fall = table_relationship.fetchall()
+    assert len(fall) == total_records
+    assert {item[0] for item in fall} == set(range(total_records))
+
+    # check fetchmany
+    many = table_relationship.fetchmany(chunk_size)
+    assert len(many) == chunk_size
+
+    # check iterfetchmany
+    chunks = list(table_relationship.iter_fetchmany(chunk_size=chunk_size))
+    assert [len(chunk) for chunk in chunks] == expected_chunk_counts
+    ids = reduce(lambda a, b: a + b, [[item[0] for item in chunk] for chunk in chunks])
+    assert set(ids) == set(range(total_records))
+
+    # simple check that query also works
+    tname = pipeline.sql_client().make_qualified_table_name("items")
+    query_relationship = pipeline.dataset().query(f"select * from {tname} where id < 20")
+
+    # we selected the first 20
+    table = query_relationship.arrow()
+    assert table.num_rows == 20
+
+    # check that hints are carried over to arrow table
```

Comment:
I think testing one column should be enough to test wether hints are carried over, we are testing all datatypes for all destinations in a few other locations